configs/components/_base-module.rb

```diff
@@ -0,0 +1,71 @@
+# This file is a basis for vendor modules that should be installed alongside
+# puppet-agent in settings[:module_vendordir]. It should not be used as a
+# component; Instead, other module components should load it using
+# `instance_eval`. Here's an example:
+#
+# component "module-puppetlabs-stdlib" do |pkg, settings, platform|
+#   pkg.version "4.25.1"
+#
+#   instance_eval File.read("configs/components/_base-module.rb")
+# end
+#
+# The module's author and name can be inferred from the name of the component
+# when it has the format "module-<author>-<name>", but you may set them
+# explicitly instead by defining `module_author` and `module_name` before
+# loading this file.
+#
+# No dependency resolution is attempted automatically; You must define separate
+# vanagon components and make their dependencies explicit.
```

Comment:
We could automatically download and install dependencies, since the forge API will tell us about them, but I've left it all manual for now. Let me know if that's something we want.

Reply:
I think manual is good for now